### PR TITLE
[Agent] fix bugs about windows

### PR DESCRIPTION
**Pull request by TomatoMr**
### This PR is for:

- Agent

#### Steps to reproduce the bug
- 
#### Changes to fix the bug
- fix debugger write beacon to client error
- fix the panic problem when failed getting control ip and mac
- fix the enabled switch bug
#### Affected branches
- main
#### Checklist
- [ ] Added unit test to verify the fix.

### CODE REVIEW

File: agent/src/utils/net/mod.rs [hunk 1]
```diff
@@ -303,6 +306,8 @@ pub fn get_ctrl_ip_and_mac(dest: IpAddr) -> (IpAddr, MacAddr) {
             let tuple = get_route_src_ip_and_mac(&dest);
```
Comment (rvql) [inline agent/src/utils/net/mod.rs:306]: 301行的要改吗

File: agent/src/trident.rs [hunk 2]
```diff
@@ -404,6 +404,9 @@ impl Trident {
                     components.replace(comp);
                 }
                 Some(components) => {
+                    if !&components.running.load(Ordering::Relaxed) {
```
Comment (rvql) [inline agent/src/trident.rs:407]: start里已经做了检查

File: agent/src/utils/net/mod.rs (hunk 1 above)
Comment (TomatoMr) [reply agent/src/utils/net/mod.rs:306]: 嗯嗯，没注意到，也要改。这里也是会panic

File: agent/src/trident.rs (hunk 2 above)
Comment (TomatoMr) [reply agent/src/trident.rs:407]: 在这里再判断一下是因为，禁用后重新启用，没有去触发components的start

Comment (rvql) [reply agent/src/trident.rs:407]: 我的意思是不需要再检查running了，直接调用start

Comment (TomatoMr) [reply agent/src/trident.rs]: 好的，已修改

**Review by rvql — APPROVED** [review]

**Review by sharang — APPROVED** [review]

**Pull request merged**